src/StoredEventRepository.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Spatie\EventProjector;
+
+use Illuminate\Support\Collection;
+use Spatie\EventProjector\Models\StoredEvent;
+
+interface StoredEventRepository
+{
+    public function retrieveAll(string $uuid = null, int $startingFrom = null): Collection;
```

Comment:
Maybe it would be cleared to split this up in 

`retrieveAll(string $uuid)` and `retrieveAllStartingFrom(int $startingFrom, string $uuid)`

If there are lots of events the `Collection` returned from these methods could become quite big. I think we should support retrieving these events in a chunked way. Maybe we could do some fancy stuff with returning a `Generator`.

When we do this, it might be harder to get a count of the events before retrieving them. We could add `countEvents(string $uuid = null)` and `countEventsStartingFrom(int $startingFrom, string $uuid)` to the interface to solve this. A database implementation could then execute a `count` query.

Reply:
Since this PR targets Laravel 6, could we do something with this? https://github.com/laravel/framework/pull/29415

#cuttingedge 🔪

Reply:
- Split up the `retrieveAll`into `retrieveAll` and `retrieveAllStartingFrom`

The 2 other points might be solved using the new LazyCollection?